### PR TITLE
Add support for complex entries

### PR DESCRIPTION
**Pull request by aradi**
Supports complex values as introduced in dftbplus/dftbplus#1208

### CODE REVIEW

**Review by bhourahine — APPROVED** [review]
LGTM, but I guess the merge should be after the main DFTB+ one.

File: src/hsd/dict.py [hunk 1]
```diff
@@ -13,24 +13,49 @@
...
+    (?:$|(?=\s+))
+)
+|
+# Complex value
+(?:
+    \s*(?:^|(?<=\s))
+    (?P<complex>
+        [-+]?[0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?i
+        |
+        [-+]?[0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?[-+][0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?i
```
Comment (jubich) [inline src/hsd/dict.py:42]:
```suggestion
    (?P<complex>
        [-+]?[0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?[ij]
        |
        [-+]?[0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?[-+][0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?[ij]
```

**Review by jubich — APPROVED** [review]
Looks good. However, since we are dealing with numerical calculations/programming languages, it might be more intuitive to use "j" instead of "i", so I would allow that as well.

File: src/hsd/dict.py (hunk 1 above)
Comment (jubich) [reply src/hsd/dict.py:42]: Well, I just saw that you specifically mentioned in dftb+ that "j" is invalid, so I guess you can ignore the comment then. ;)

**Comment by aradi** [comment]
@jubich Thx for the review! Actually, I had first allowed the `j` suffix as an alternative form (also in DFTB+), but then took it out so that we have only one unique way to write imaginary units now. We can later still allow for `j`, if really needed. Anyway, Python is probably the only programming language with complex numbers, which does not allow for using `i` or `im` as imaginary suffix.

**Pull request merged**